bitnami/kafka/templates/kafka-provisioning-secret.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.provisioning.enabled (include "kafka.client.tlsEncryption" .) (not .Values.provisioning.auth.tls.passwordsSecret)  }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "kafka.client.passwordsSecretName" . }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: Opaque
+data:
+  {{- if .Values.provisioning.auth.tls.keystorePassword }}
+  truststore-password: {{ default "" .Values.provisioning.auth.tls.keystorePassword | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.provisioning.auth.tls.truststorePassword }}
+  keystore-password: {{ default "" .Values.provisioning.auth.tls.truststorePassword | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.provisioning.auth.tls.keyPassword }}
+  key-password: {{ default "" .Values.provisioning.auth.tls.keyPassword | b64enc | quote }}
+  {{- end }}
```

Comment:
```suggestion
  truststore-password: {{ default "" .Values.provisioning.auth.tls.keystorePassword | b64enc | quote }}
  keystore-password: {{ default "" .Values.provisioning.auth.tls.truststorePassword | b64enc | quote }}
  key-password: {{ default "" .Values.provisioning.auth.tls.keyPassword | b64enc | quote }}
```

Remove conditionals. Otherwise, this section will cause the pod to fail because of "Secret missing key" error.

```

            {{- if (include "kafka.client.tlsEncryption" .) }}
            - name: KAFKA_CLIENT_KEY_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ template "kafka.client.passwordsSecretName" . }}
                  key: {{ .Values.provisioning.auth.tls.keyPasswordSecretKey }}
            - name: KAFKA_CLIENT_KEYSTORE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ template "kafka.client.passwordsSecretName" . }}
                  key: {{ .Values.provisioning.auth.tls.keystorePasswordSecretKey }}
            - name: KAFKA_CLIENT_TRUSTSTORE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ template "kafka.client.passwordsSecretName" . }}
                  key: {{ .Values.provisioning.auth.tls.truststorePasswordSecretKey }}
            {{- end }}
```

Since they default to "", the container logic won't take them under consideration if empty.